chirp/drivers/mml_jc8810.py

```diff
@@ -0,0 +1,1070 @@
+    return True
+
+
+class JC8810base(chirp_common.CloneModeRadio):
+    """MML JC-8810"""
+    VENDOR = "MML"
+    MODEL = "JC-8810base"
+    BAUD_RATE = 57600
+    NEEDS_COMPAT_SERIAL = False
+    BLOCK_SIZE = 0x40
+    BLOCK_SIZE_UP = 0x40
+
+    POWER_LEVELS = [chirp_common.PowerLevel("H", watts=10.00),
+                    chirp_common.PowerLevel("M", watts=8.00),
+                    chirp_common.PowerLevel("L", watts=4.00)]
+
+    _magic = b"PROGRAMJC81U"
+    _fingerprint = [b"\x00\x00\x00\x26\x00\x20\xD8\x04",
+                    b"\x00\x00\x00\x42\x00\x20\xF0\x04"]
+
+    _ranges = [
+               (0x0000, 0x2000),
+               (0x8000, 0x8040),
+               (0x9000, 0x9040),
+               (0xA000, 0xA140),
+               (0xB000, 0xB300)
+              ]
+    _memsize = 0xB300
+    _valid_chars = chirp_common.CHARSET_ALPHANUMERIC + \
+        "`~!@#$%^&*()-=_+[]\\{}|;':\",./<>?"
+
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.has_settings = True
+        rf.has_bank = False
+        rf.has_ctone = True
+        rf.has_cross = True
+        rf.has_rx_dtcs = True
+        rf.has_tuning_step = False
+        rf.can_odd_split = True
+        rf.has_name = True
+        rf.valid_name_length = 12
+        rf.valid_characters = self._valid_chars
+        rf.valid_skips = ["", "S"]
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+        rf.valid_cross_modes = ["Tone->Tone", "Tone->DTCS", "DTCS->Tone",
+                                "->Tone", "->DTCS", "DTCS->", "DTCS->DTCS"]
+        rf.valid_power_levels = self.POWER_LEVELS
+        rf.valid_duplexes = ["", "-", "+", "split", "off"]
+        rf.valid_modes = ["FM", "NFM"]  # 25 kHz, 12.5 KHz.
+        rf.valid_dtcs_codes = DTCS
+        rf.memory_bounds = (1, 256)
+        rf.valid_tuning_steps = [2.5, 5., 6.25, 10., 12.5, 20., 25., 50.]
+        rf.valid_bands = [(108000000, 136000000),
+                          (136000000, 180000000),
+                          (200000000, 260000000),
+                          (330000000, 400000000),
+                          (400000000, 520000000)]
+        return rf
+
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    def sync_in(self):
+        """Download from radio"""
+        try:
+            data = do_download(self)
+        except errors.RadioError:
+            # Pass through any real errors we raise
+            raise
+        except Exception:
+            # If anything unexpected happens, make sure we raise
+            # a RadioError and log the problem
+            LOG.exception('Unexpected error during download')
+            raise errors.RadioError('Unexpected error communicating '
+                                    'with the radio')
+        self._mmap = data
+        self.process_mmap()
+
+    def sync_out(self):
+        """Upload to radio"""
+        try:
+            do_upload(self)
+        except Exception:
```

Comment:
It'd be better to pass through any `RadioError` here like you do for `sync_in()` so you get the better error reporting if it happens.